Verbatim strings produce incorrect results

**Describe the bug**
Incorrect behavior when dealing with verbatim strings in languages such as C++, Python, and Rust

**To Reproduce**

**Python**:
```python
text = r"xyz\\"
# hi
```
**Output**:
```text
───────────────────────────────────────────────────────────────────────────────
Language                 Files     Lines   Blanks  Comments     Code Complexity
───────────────────────────────────────────────────────────────────────────────
Python                       1         2        0         0        2          0
───────────────────────────────────────────────────────────────────────────────
Total                        1         2        0         0        2          0
───────────────────────────────────────────────────────────────────────────────
Estimated Cost to Develop $39
Estimated Schedule Effort 0.325843 months
Estimated People Required 0.014395
───────────────────────────────────────────────────────────────────────────────
```
**C++**:
```cpp
/* 46 lines 37 code 3 comments 6 blanks */

#include <stdio.h>

// bubble_sort_function
void bubble_sort(int a[10], int n) {
  int t;
  int j = n;
  int s = 1;
  while (s > 0) {
    s = 0;
    int i = 1;
    while (i < j) {
      if (a[i] < a[i - 1]) {
        t = a[i];
        a[i] = a[i - 1];
        a[i - 1] = t;
        s = 1;
      }
      i++;
    }
    j--;
  }
}

int main() {
  int a[] = {4, 65, 2, -31, 0, 99, 2, 83, 782, 1};
  int n = 10;
  int i = 0;

  printf(R"(Before sorting:\n\n" )");
  // Single line comment
  while (i < n) {
    printf("%d ", a[i]);
    i++;
  }

  bubble_sort(a, n);

  printf("\n\nAfter sorting:\n\n");
  i = 0;
  while (i < n) {
    printf("%d ", a[i]);
    i++;
  }
}
```
This is the example [Tokei](https://github.com/XAMPPRocky/tokei/blob/master/tests/data/cpp.cpp) uses.
**Output**:
```text
───────────────────────────────────────────────────────────────────────────────
Language                 Files     Lines   Blanks  Comments     Code Complexity
───────────────────────────────────────────────────────────────────────────────
C++                          1        46        4         2       40          3
───────────────────────────────────────────────────────────────────────────────
Total                        1        46        4         2       40          3
───────────────────────────────────────────────────────────────────────────────
Estimated Cost to Develop $919
Estimated Schedule Effort 1.076760 months
Estimated People Required 0.101203
───────────────────────────────────────────────────────────────────────────────
```
**Rust**:
```rust
// 41 lines 33 code 3 comments 5 blanks

/* /**/ */
fn main() {
    let start = r##"/*##\"
\"##;
    // comment
    loop {
        if x.len() >= 2 && x[0] == '*' && x[1] == '/' { // found the */
            break;
        }
    }
}

fn foo<'a, 'b>(name: &'b str) {
    let this_ends = "a \"test/*.";
    call1();
    call2();
    let this_does_not = /* a /* nested */ comment " */
        "*/another /*test
            call3();
            */";
}

fn foobar() {
    let does_not_start = // "
        "until here,
        test/*
        test"; // a quote: "
    let also_doesnt_start = /* " */
        "until here,
        test,*/
        test"; // another quote: "
}

fn foo() {
    let a = 4; // /*
    let b = 5;
    let c = 6; // */
}

```
This is the example [Tokei](https://github.com/XAMPPRocky/tokei/blob/master/tests/data/rust.rs) uses.
**Output**:
```text
───────────────────────────────────────────────────────────────────────────────
Language                 Files     Lines   Blanks  Comments     Code Complexity
───────────────────────────────────────────────────────────────────────────────
Rust                         1        41        3         6       32          0
───────────────────────────────────────────────────────────────────────────────
Total                        1        41        3         6       32          0
───────────────────────────────────────────────────────────────────────────────
Estimated Cost to Develop $727
Estimated Schedule Effort 0.985035 months
Estimated People Required 0.087520
───────────────────────────────────────────────────────────────────────────────
```

**Expected behavior**
**Python**:
2 lines 1 comment 1 code 0 blanks
**C++**:
46 lines 3 comment 37 code 6 blanks
**Rust**:
41 lines 33 code 3 comments 5 blanks

This is the default case, as far as I know for C++ verbatim strings, because [C++ verbatim strings can have prefixes](http://stroustrup.com/C++11FAQ.html#raw-strings).

Rust supports the pattern of `r(#+)"..."(#+)` where the capture groups must have equal length.

**Desktop (please complete the following information):**
 - OS: Linux
 - Version: 2.12.0

Love the project :heart:

Man that c++ file is screwy... 

```
# bb100123 @ VCOANSYD256197 in ~ [11:35:34]
$ gcc test.cpp
test.cpp:31:10: error: use of undeclared identifier 'R'
  printf(R"(Before sorting:\n\n" )");
         ^
test.cpp:31:35: warning: missing terminating '"' character [-Winvalid-pp-token]
  printf(R"(Before sorting:\n\n" )");
                                  ^
1 warning and 1 error generated.

# bb100123 @ VCOANSYD256197 in ~ [11:35:36] C:1
$ clang
clang: error: no input files

# bb100123 @ VCOANSYD256197 in ~ [11:35:45] C:1
$ clang test.cpp
test.cpp:31:10: error: use of undeclared identifier 'R'
  printf(R"(Before sorting:\n\n" )");
         ^
test.cpp:31:35: warning: missing terminating '"' character [-Winvalid-pp-token]
  printf(R"(Before sorting:\n\n" )");
                                  ^
1 warning and 1 error generated.
```
Is there some trick to get that to work? The issue I see with it is,

```
printf(R"(Before sorting:\n\n" )");
```
Where it looks like the string ends after the \n\n as its not escaped?

--- a/NOTES.md
+++ b/NOTES.md
@@ -34,6 +34,12 @@ These can change scc's output or behaviour for an identical invocation against
 - **Percentage outputs in JSON** (#720) — JSON output now includes percentage
   figures.
 - **Codeberg support** (#866376f) — Codeberg as a supported badge/target.
+- **External ignore files via `--ignore-file`** — supply additional gitignore-format
+  ignore files from outside the scanned tree (repeatable, order-sensitive, anchored
+  at the scan root; any in-tree `.gitignore`/`.ignore`/`.sccignore` takes precedence).
+  Resolves the long-standing request for global/shared ignore support without scc
+  shelling out to git — point it at `~/.config/git/ignore` directly, or set it once
+  in a `.scc` file via `SCC_CONFIG_PATH`. Backed by `CustomIgnoreFiles` in gocodewalker.
 
 ## 🗣 Language support
 
@@ -63,6 +69,15 @@ These can change scc's output or behaviour for an identical invocation against
 - Apply `go fix` suggestions (#712); convert more shell tests to Go tests (#696).
 - Automate file generation and add CI tests (#688, #706); update to latest Go
   and refresh dependencies.
+
+## 🗒 To discuss / follow-up
+
+- **Remaining `git` subprocess shell-out** — `remoteOriginName` in
+  `processor/report.go` still runs `git config --get remote.origin.url` to derive
+  the HTML report title. This is the only place scc shells out to the `git` binary
+  (the rest use the go-git library or `gocodewalker.FindRepositoryRoot`). Decide
+  whether to drop it (fall back to the path basename) or read it via go-git's config
+  instead, so scc has no hard dependency on a `git` binary being installed.
 
 ---
 

--- a/SCC-OUTPUT-REPORT.html
+++ b/SCC-OUTPUT-REPORT.html
@@ -12,24 +12,24 @@
 	</tr></thead>
 	<tbody><tr>
 		<th>Go</th>
-		<th>65</th>
-		<th>39405</th>
-		<th>2953</th>
-		<th>1963</th>
-		<th>34489</th>
-		<th>4336</th>
-		<th>940656</th>
-		<th>14313</th>
+		<th>66</th>
+		<th>39658</th>
+		<th>2965</th>
+		<th>1976</th>
+		<th>34717</th>
+		<th>4352</th>
+		<th>946507</th>
+		<th>14376</th>
 	</tr><tr>
 		<td>processor/constants.go</td>
 		<td></td>
-		<td>14687</td>
+		<td>14855</td>
 		<td>1</td>
 		<td>2</td>
-		<td>14684</td>
+		<td>14852</td>
 		<td>0</td>
-		<td>235991</td>
-		<td>2249</td>
+		<td>239226</td>
+		<td>2280</td>
 	</tr><tr>
 		<td>processor/workers_test.go</td>
 		<td></td>
@@ -53,13 +53,13 @@
 	</tr><tr>
 		<td>processor/processor.go</td>
 		<td></td>
-		<td>1042</td>
-		<td>197</td>
-		<td>188</td>
-		<td>657</td>
+		<td>1049</td>
+		<td>198</td>
+		<td>192</td>
+		<td>659</td>
 		<td>156</td>
-		<td>31732</td>
-		<td>681</td>
+		<td>32102</td>
+		<td>687</td>
 	</tr><tr>
 		<td>main_test.go</td>
 		<td></td>
@@ -83,23 +83,23 @@
 	</tr><tr>
 		<td>processor/workers.go</td>
 		<td></td>
-		<td>957</td>
+		<td>960</td>
 		<td>146</td>
-		<td>95</td>
+		<td>98</td>
 		<td>716</td>
 		<td>244</td>
-		<td>28545</td>
-		<td>543</td>
+		<td>28788</td>
+		<td>547</td>
 	</tr><tr>
 		<td>processor/report.go</td>
 		<td></td>
-		<td>805</td>
+		<td>806</td>
 		<td>79</td>
 		<td>108</td>
-		<td>618</td>
+		<td>619</td>
 		<td>112</td>
-		<td>22568</td>
-		<td>580</td>
+		<td>22613</td>
+		<td>581</td>
 	</tr><tr>
 		<td>processor/history.go</td>
 		<td></td>
@@ -113,13 +113,13 @@
 	</tr><tr>
 		<td>config_test.go</td>
 		<td></td>
-		<td>785</td>
+		<td>786</td>
 		<td>50</td>
 		<td>85</td>
-		<td>650</td>
+		<td>651</td>
 		<td>199</td>
-		<td>29332</td>
-		<td>410</td>
+		<td>29349</td>
+		<td>411</td>
 	</tr><tr>
 		<td>processor/report_render.go</td>
 		<td></td>
@@ -173,13 +173,13 @@
 	</tr><tr>
 		<td>config.go</td>
 		<td></td>
-		<td>584</td>
+		<td>585</td>
 		<td>24</td>
 		<td>49</td>
-		<td>511</td>
+		<td>512</td>
 		<td>40</td>
-		<td>27743</td>
-		<td>420</td>
+		<td>27979</td>
+		<td>421</td>
 	</tr><tr>
 		<td>processor/history_author_timeline_test.go</td>
 		<td></td>
@@ -251,16 +251,6 @@
 		<td>10674</td>
 		<td>281</td>
 	</tr><tr>
-		<td>processor/file_test.go</td>
-		<td></td>
-		<td>373</td>
-		<td>70</td>
-		<td>16</td>
-		<td>287</td>
-		<td>75</td>
-		<td>8978</td>
-		<td>178</td>
-	</tr><tr>
 		<td>processor/history_languages.go</td>
 		<td></td>
 		<td>373</td>
@@ -270,6 +260,16 @@
 		<td>52</td>
 		<td>10482</td>
 		<td>269</td>
+	</tr><tr>
+		<td>processor/file_test.go</td>
+		<td></td>
+		<td>373</td>
+		<td>70</td>
+		<td>16</td>
+		<td>287</td>
+		<td>75</td>
+		<td>8978</td>
+		<td>178</td>
 	</tr><tr>
 		<td>processor/formatters_misc.go</td>
 		<td></td>
@@ -503,13 +503,13 @@
 	</tr><tr>
 		<td>processor/result.go</td>
 		<td></td>
-		<td>137</td>
+		<td>138</td>
 		<td>24</td>
 		<td>3</td>
-		<td>110</td>
+		<td>111</td>
 		<td>31</td>
-		<td>3247</td>
-		<td>92</td>
+		<td>3291</td>
+		<td>93</td>
 	</tr><tr>
 		<td>processor/formatters_sql.go</td>
 		<td></td>
@@ -601,6 +601,16 @@
 		<td>2123</td>
 		<td>62</td>
 	</tr><tr>
+		<td>processor/workers_issue175_test.go</td>
+		<td></td>
+		<td>71</td>
+		<td>11</td>
+		<td>6</td>
+		<td>54</td>
+		<td>16</td>
+		<td>1661</td>
+		<td>37</td>
+	</tr><tr>
 		<td>processor/history_ignore_test.go</td>
 		<td></td>
 		<td>55</td>
@@ -673,16 +683,16 @@
 	</tr></tbody>
 	<tfoot><tr>
 		<th>Total</th>
-		<th>65</th>
-		<th>39405</th>
-		<th>2953</th>
-		<th>1963</th>
-		<th>34489</th>
-		<th>4336</th>
-		<th>940656</th>
-		<th>14313</th>
+		<th>66</th>
+		<th>39658</th>
+		<th>2965</th>
+		<th>1976</th>
+		<th>34717</th>
+		<th>4352</th>
+		<th>946507</th>
+		<th>14376</th>
 	</tr>
 	<tr>
-		<th colspan="9">Estimated Cost to Develop (organic) $1,112,141<br>Estimated Schedule Effort (organic) 14.32 months<br>Estimated People Required (organic) 6.90<br></th>
+		<th colspan="9">Estimated Cost to Develop (organic) $1,119,862<br>Estimated Schedule Effort (organic) 14.36 months<br>Estimated People Required (organic) 6.93<br></th>
 	</tr></tfoot>
 	</table></body></html>

--- a/languages.json
+++ b/languages.json
@@ -1042,6 +1042,31 @@
     "multi_line": [["/*", "*/"]],
     "quotes": [
       {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "R\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "u8R\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "uR\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "UR\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "LR\"("
+      },
+      {
         "end": "\"",
         "start": "\""
       }
@@ -1082,6 +1107,31 @@
     "line_comment": ["//"],
     "multi_line": [["/*", "*/"]],
     "quotes": [
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "R\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "u8R\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "uR\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "UR\"("
+      },
+      {
+        "ignoreEscape": true,
+        "end": ")\"",
+        "start": "LR\"("
+      },
       {
         "end": "\"",
         "start": "\""
@@ -6458,6 +6508,96 @@
     "multi_line": [["/*", "*/"]],
     "nestedmultiline": true,
     "quotes": [
+      {
+        "ignoreEscape": true,
+        "end": "\"",
+        "start": "r\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"#",
+        "start": "r#\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"##",
+        "start": "r##\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"###",
+        "start": "r###\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"####",
+        "start": "r####\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"#####",
+        "start": "r#####\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"######",
+        "start": "r######\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"#######",
+        "start": "r#######\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"########",
+        "start": "r########\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"",
+        "start": "br\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"#",
+        "start": "br#\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"##",
+        "start": "br##\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"###",
+        "start": "br###\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"####",
+        "start": "br####\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"#####",
+        "start": "br#####\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"######",
+        "start": "br######\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"#######",
+        "start": "br#######\""
+      },
+      {
+        "ignoreEscape": true,
+        "end": "\"########",
+        "start": "br########\""
+      },
       {
         "end": "\"",
         "start": "\""

--- a/processor/constants.go
+++ b/processor/constants.go
@@ -1720,6 +1720,36 @@ var languageDatabase = map[string]Language{
 		},
 		Quotes: []Quote{
 			{
+				Start:        "R\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "u8R\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "uR\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "UR\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "LR\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
 				Start:        "\"",
 				End:          "\"",
 				IgnoreEscape: false,
@@ -1775,6 +1805,36 @@ var languageDatabase = map[string]Language{
 			},
 		},
 		Quotes: []Quote{
+			{
+				Start:        "R\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "u8R\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "uR\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "UR\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "LR\"(",
+				End:          ")\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
 			{
 				Start:        "\"",
 				End:          "\"",
@@ -10680,6 +10740,114 @@ var languageDatabase = map[string]Language{
 			},
 		},
 		Quotes: []Quote{
+			{
+				Start:        "r\"",
+				End:          "\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r#\"",
+				End:          "\"#",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r##\"",
+				End:          "\"##",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r###\"",
+				End:          "\"###",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r####\"",
+				End:          "\"####",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r#####\"",
+				End:          "\"#####",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r######\"",
+				End:          "\"######",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r#######\"",
+				End:          "\"#######",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "r########\"",
+				End:          "\"########",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br\"",
+				End:          "\"",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br#\"",
+				End:          "\"#",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br##\"",
+				End:          "\"##",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br###\"",
+				End:          "\"###",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br####\"",
+				End:          "\"####",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br#####\"",
+				End:          "\"#####",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br######\"",
+				End:          "\"######",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br#######\"",
+				End:          "\"#######",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
+			{
+				Start:        "br########\"",
+				End:          "\"########",
+				IgnoreEscape: true,
+				DocString:    false,
+			},
 			{
 				Start:        "\"",
 				End:          "\"",

--- a/processor/workers.go
+++ b/processor/workers.go
@@ -239,7 +239,10 @@ func stringState(fileJob *FileJob, index int, endPoint int, endString []byte, cu
 		// If we are in a literal string we want to ignore escapes OR we aren't checking for special ones
 		if ignoreEscape || !is_escaped {
 			if checkForMatchSingle(fileJob.Content[i], index, endPoint, endString, fileJob) {
-				return i, SCode
+				// Skip past the whole end delimiter. For multi byte terminators such
+				// as the C++ raw string )" the trailing byte is itself a quote start,
+				// so leaving the cursor on it would re-open a new string. See #175.
+				return i + len(endString) - 1, SCode
 			}
 		}
 	}

--- a/processor/workers_issue175_test.go
+++ b/processor/workers_issue175_test.go
@@ -1,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+
+package processor
+
+import (
+	"testing"
+)
+
+// https://github.com/boyter/scc/issues/175
+// Raw/verbatim strings contain characters that look like string terminators,
+// comment starts or escapes. Without dedicated quote rules the embedded "
+// prematurely closes the string, the leftover quote opens a phantom string and
+// the real comment that follows is swallowed and counted as code.
+
+func TestCountStatsIssue175CPlusPlusRawString(t *testing.T) {
+	ProcessConstants()
+	fileJob := FileJob{
+		Language: "C++",
+	}
+
+	fileJob.SetContent(`/* 6 lines 3 code 2 comments 1 blanks */
+int main() {
+  const char* s = R"(has a " and \n inside)";
+
+  // a real comment
+}`)
+
+	CountStats(&fileJob)
+
+	if fileJob.Lines != 6 {
+		t.Errorf("Expected 6 lines got %d", fileJob.Lines)
+	}
+	if fileJob.Code != 3 {
+		t.Errorf("Expected 3 code got %d", fileJob.Code)
+	}
+	if fileJob.Comment != 2 {
+		t.Errorf("Expected 2 comments got %d", fileJob.Comment)
+	}
+	if fileJob.Blank != 1 {
+		t.Errorf("Expected 1 blank got %d", fileJob.Blank)
+	}
+}
+
+func TestCountStatsIssue175RustRawString(t *testing.T) {
+	ProcessConstants()
+	fileJob := FileJob{
+		Language: "Rust",
+	}
+
+	fileJob.SetContent(`// 6 lines 3 code 2 comments 1 blanks
+fn main() {
+    let s = r#"has a " and /* inside"#;
+
+    // a real comment
+}`)
+
+	CountStats(&fileJob)
+
+	if fileJob.Lines != 6 {
+		t.Errorf("Expected 6 lines got %d", fileJob.Lines)
+	}
+	if fileJob.Code != 3 {
+		t.Errorf("Expected 3 code got %d", fileJob.Code)
+	}
+	if fileJob.Comment != 2 {
+		t.Errorf("Expected 2 comments got %d", fileJob.Comment)
+	}
+	if fileJob.Blank != 1 {
+		t.Errorf("Expected 1 blank got %d", fileJob.Blank)
+	}
+}

